source/concurrent/PartiallyGuardedList.h

```diff
@@ -0,0 +1,61 @@
+/* PartiallyGuardedList.h
+Copyright (c) 2024 by tibetiroka
+
+Endless Sky is free software: you can redistribute it and/or modify it under the
+terms of the GNU General Public License as published by the Free Software
+Foundation, either version 3 of the License, or (at your option) any later version.
+
+Endless Sky is distributed in the hope that it will be useful, but WITHOUT ANY
+WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+PARTICULAR PURPOSE. See the GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License along with
+this program. If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#ifndef PARTIALLY_GUARDED_LIST_H_
+#define PARTIALLY_GUARDED_LIST_H_
+
+#include <list>
+#include <mutex>
+
+
+
+// Vectors that guard against most concurrent modifications. Use with caution.
+// For most purposes, this can be treated and passed around like any stl container,
+// but any function modifying it inside a concurrent context MUST receive it as a PartiallyGuarded object
+// to ensure proper handling.
+// Iterators, clear(), swap() and some other functions are left unguarded, as they are generally not used
+// for concurrent modifications.
+template<class T, class Allocator = std::allocator<T>>
+class PartiallyGuardedList : public std::list<T, Allocator> {
+public:
+	template<class... Args>
+	inline T &emplace_back(Args &&... args)
+	{
+		const std::lock_guard<std::mutex> lock(write_mutex);
+		return std::list<T, Allocator>::emplace_back(args...);
+	}
```

Comment:
ah just noticed this now, no can do, move downwards according to style guide

Reply:
I thought small inline functions were allowed here?

Reply:
Nope